app/init/launch.ts

```diff
@@ -67,13 +67,18 @@ const launchAppFromNotification = async (notification: NotificationWithData, col
 
  * @returns a redirection to a screen, either onboarding, add_server, login or home depending on the scenario
  */
-const launchApp = async (props: LaunchProps, resetNavigation = true) => {
+const launchApp = async (props: LaunchProps) => {
     let serverUrl: string | undefined;
     switch (props?.launchType) {
         case Launch.DeepLink:
             if (props.extra?.type !== DeepLink.Invalid) {
                 const extra = props.extra as DeepLinkWithData;
-                serverUrl = extra.data?.serverUrl;
+                const existingServer = DatabaseManager.searchUrl(extra.data!.serverUrl);
+                serverUrl = existingServer;
+                props.serverUrl = serverUrl || extra.data?.serverUrl;
```

Comment:
* Lines can be compressed.  
* `existingServer` is redundant.  
* `extra.data!.serverUrl` - here serverUrl can be undefined right?

Reply:
Yes, it can be undefined, thus normal launch is expected